test_scripts/Polices/External_UCS/016_ATF_N_TC_PTU_with_external_consent_status_groups_struct.lua

```diff
@@ -0,0 +1,127 @@
+-- 3. Activate app
+-- 4. Perform PTU
+-- 5. Verify status of update
+--
+-- Expected result:
+-- PTU fails with UPDATE_NEEDED status
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local json = require("modules/json")
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+
+--[[ Local variables ]]
+local appId = config.application1.registerAppInterfaceParams.appID
+local grpId = "Location-1"
+local checkedStatus = "UPDATE_NEEDED"
+
+--[[ Local Functions ]]
+local function replaceSDLPreloadedPtFile()
+  local preloadedFile = commonPreconditions:GetPathToSDL() ..
+  commonFunctions:read_parameter_from_smart_device_link_ini("PreloadedPT")
+  local preloadedTable = testCasesForExternalUCS.createTableFromJsonFile(preloadedFile)
+  preloadedTable.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  --
+  preloadedTable.policy_table.app_policies[appId] = {
+    default_hmi = "NONE",
+    keep_context = false,
+    priority = "NONE",
+    steal_focus = false,
+    groups = {
+      "Base-4", grpId
+    }
+  }
+  preloadedTable.policy_table.functional_groupings[grpId].disallowed_by_external_consent_entities_on = {
+    {
+      entityID = 128,
+      entityType = 0
+    }
+  }
+  testCasesForExternalUCS.createJsonFileFromTable(preloadedTable, preloadedFile)
+end
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonPreconditions:BackupFile("sdl_preloaded_pt.json")
+replaceSDLPreloadedPtFile()
+testCasesForExternalUCS.removePTS()
+
+--[[ General Settings for configuration ]]
+Test = require("user_modules/connecttest_resumption")
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:ConnectMobile()
+  self:connectMobile()
+end
+
+function Test:StartSession()
+  testCasesForExternalUCS.startSession(self, 1)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:RAI()
+  testCasesForExternalUCS.registerApp(self, 1)
+end
+
+function Test:SendExternalConsent()
+  self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent", { source = "GUI",
+      externalConsentStatus = {
+        { entityType = 0, entityID = 128, status = "OFF" }
+    } })
+end
+
+function Test:ActivateApp()
+  testCasesForExternalUCS.activateApp(self, 1, checkedStatus)
```

Comment:
@dboltovskyi, I didn't manage to find where you update ptu file with disallowed_by_external_consent_entities_on.
Please clarify.